frontend/src/components/BookTable.tsx

```diff
@@ -0,0 +1,624 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import {
+  Box,
+  Typography,
+  Paper,
+  Stack,
+  ListItemButton,
+  ListItemText,
+  ListItemAvatar,
+  useTheme,
+  CircularProgress,
+} from '@mui/material';
+import { DataGrid } from '@mui/x-data-grid';
+import currencyDict from '../../static/assets/currencies.json';
+import { Order } from '../models/Order.model';
+
+import FlagWithProps from './FlagWithProps';
+import { pn, amountToString } from '../utils/prettyNumbers';
+import PaymentText from './PaymentText';
+import RobotAvatar from './Robots/RobotAvatar';
+
+function statusBadgeColor(status) {
+  if (status === 'Active') {
+    return 'success';
+  }
+  if (status === 'Seen recently') {
+    return 'warning';
+  }
+  if (status === 'Inactive') {
+    return 'error';
+  }
+}
+
+function hexToRgb(c: string) {
+  if (c.includes('rgb')) {
+    const vals = c.split('(')[1].split(')')[0];
+    return vals.split(',');
+  }
+  if (/^#([a-f0-9]{3}){1,2}$/.test(c)) {
+    if (c.length == 4) {
+      c = '#' + [c[1], c[1], c[2], c[2], c[3], c[3]].join('');
+    }
+    c = '0x' + c.substring(1);
+    return [(c >> 16) & 255, (c >> 8) & 255, c & 255];
+  }
+  return '';
+}
+
+function mixColors(baseColor: string, accentColor: string, point: number) {
+  const baseRGB = hexToRgb(baseColor);
+  const accentRGB = hexToRgb(accentColor);
+  const redDiff = accentRGB[0] - baseRGB[0];
+  const red = baseRGB[0] + redDiff * point;
+  const greenDiff = accentRGB[1] - baseRGB[1];
+  const green = baseRGB[1] + greenDiff * point;
+  const blueDiff = accentRGB[2] - baseRGB[2];
+  const blue = baseRGB[2] + blueDiff * point;
+  return `rgb(${Math.round(red)}, ${Math.round(green)}, ${Math.round(blue)}, ${0.7 + point * 0.3})`;
+}
```

Comment:
Maybe we can move these functions to utils?

Reply:
Done!